### ISSUE TITLE
FAC - Bug: Incomplete Lookup Lists

### ISSUE THREAD

**Issue by sambodeme** [GSA-TTS/FAC#1643]
### Describe the bug

We use [lookup lists](https://github.com/GSA-TTS/FAC/tree/main/backend/schemas/source/data) to ensure validation within worksheets and for schema validations. However, these lookup lists are not exhaustives, causing validation issues.

### Steps to reproduce the bug

1. Go to data [files](https://github.com/GSA-TTS/FAC/tree/main/backend/schemas/source/base).
2. Open ClusterNames.json and look for `CDBG - ENTITLEMENT GRANTS CLUSTER` or any variation.
3. Confirm that you only one instance of `CDBG - ENTITLEMENT GRANTS CLUSTER` with no variation.
4. Download 2022 data from [Census](https://facdissem.census.gov/PublicDataDownloads.aspx) or from Google drive in [SQLite](https://drive.google.com/drive/u/2/folders/1ETDwj4xDv_mUOHLt39e-3HmEi2Q-E5FF).
5. Open the CFDA table and look for `CDBG - ENTITLEMENT GRANTS CLUSTER` and any variation within CLUSTERNAME column.
6. Note that there are three variations: `CDBG - ENTITLEMENT GRANTS CLUSTER`, `CDBG ENTITLEMENT GRANTS CLUSTER`, and `CDBG-ENTITLEMENT GRANTS CLUSTER`. 

### Expected Behavior

The lookup lists should be exhaustive, containing all possible variations of the items they are meant to validate. When using the lookup lists for validation within worksheets or schema validations, the lists should include all valid options, ensuring that any legitimate data can be successfully validated without encountering issues. If, for some reason, we are unable to maintain the lookup lists updated, then validations should not be enforced against these lists.

### Code of Conduct

- [X] I agree to abide by the [Digital.gov Community Guidelines](https://digital.gov/communities/community-guidelines/) and the [TTS Code of Conduct](https://handbook.tts.gsa.gov/code-of-conduct/). Respect your peers, use plain language, and be patient.
- [X] I checked the [current issues](https://github.com/GSA-TTS/FAC/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by jadudm** [comment]
ALNs are a problem. So are associated program names.

1. Accept ALNs that match our regex; this should include 01-09 for agency code and the program number should include possible RD and U values.
2. Our workbook can do a `VLOOKUP` against the ALN. If we find a match in our backing sheet, we can populate the column. If not, we can say `UNKNOWN`. We do not validate the field, and therefore,  users can enter their own value over the UNKNOWN.

In short, we have no way of validating past ALNs in an authoritative manner without substantial new engineering effort, where we track ALNs over time and match them to the grant issuance date. Even then, it may not be possible to salvage integrity from the ALN.

**Comment by danswick** [comment]
@jadudm just to confirm, we've decided not to attempt to enforce canonical ALNs/program names and will instead use a combination of regex and attempt to auto-populate the "program names" column as a convenience that users can override if need be. Is that right?

**Comment by jadudm** [comment]
That is correct.

ADR to follow, to that effect.

**Comment by jadudm** [comment]
For clarity (to become  an ADR):

1. There are funding sources that do not have ALNs or program names. THEREFORE
2. The best we can do, in our workbooks, is attempt to look up the ALN, and offer a program name. HOWEVER
3. The user will need to be able to enter values for agency number, program number/three-digit extension, and program name that might not be in a "canonical" lookup list. Hence, we cannot lock these columns to strict validations.
4. We *can* use a regular expression (in our validation schema) to make sure that the ALN (or agency number combined with three-digit extension) is "valid." This includes ALNs with RD and U.

This is **not** a regression in data quality from the approach Census took. It is, as far as we can tell, the same approach that Census took. The ALN is, as we have discovered time and again, a poorly controlled/too-small number for the work it needs to do.

## To Do

1. Remove lookup lists from validations for ALN. This would include the agency numbers, etc.
2. Move the three-digit extension regex out from the validation, and give it a top-level definition for `ALNRegex`. Use that value in the validation. (If need be, break it into two parts, and combine them akin to UEI.)
3. Only use that regex for validation as opposed to lookup lists. We'll want to make sure that relevant workbook validations use the regex. (E.g. FederalAwards)
